ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/PullQueryRewriter.java

```diff
@@ -36,10 +37,16 @@ private PullQueryRewriter() {
   public static Expression rewrite(final Expression expression) {
     final Expression pseudoTimestamp = new StatementRewriteForMagicPseudoTimestamp()
         .rewrite(expression);
-    final Expression inPredicatesRemoved = rewriteInPredicates(pseudoTimestamp);
+    final Expression betweenPredicatesRemoved = rewriteBetweenPredicates(pseudoTimestamp);
+    final Expression inPredicatesRemoved = rewriteInPredicates(betweenPredicatesRemoved);
```

Comment:
We have a test class `PullQueryRewriterTest`.  Would be good to add some good test cases.

Reply:
Added tests for between predicate.